src/SpendPermissionManager.sol

```diff
@@ -279,15 +302,44 @@ contract SpendPermissionManager is EIP712 {
         _transferFrom(spendPermission.token, spendPermission.account, spendPermission.spender, value);
     }
 
+    /// @notice Revokes a spend permission and approves a new one atomically.
+    ///
+    /// @dev Enforces that the last updated period of the permission being revoked matches the last valid updated period
+    ///      submitted as an argument. This is to prevent frontrunning `replace` with additional last-minute spends.
+    ///
+    /// @dev The `account` of the permissions must match, but the remaining fields can differ.
+    ///
+    /// @dev Can only be called by the `account` of a permission.
+    ///
+    /// @param permissionToRevoke Details of the spend permission to revoke.
+    /// @param permissionToApprove Details of the spend permission to approve.
+    /// @param lastValidUpdatedPeriod Last valid updated period for the spend permission being revoked.
+    function replace(
+        SpendPermission calldata permissionToRevoke,
+        SpendPermission calldata permissionToApprove,
+        PeriodSpend calldata lastValidUpdatedPeriod
```

Comment:
What do you think of this renaming?
```suggestion
    function approveWithRevoke(
        SpendPermission calldata permissionToApprove,
        SpendPermission calldata permissionToRevoke,
        PeriodSpend calldata requiredLastUpdatedPeriod
```

Reply:
i like the emphasis on the approval that's happening